### ISSUE TITLE
RepositoryNotFoundException

### ISSUE THREAD

**Issue by zhaohailongzhl** [MovingBlocks/Terasology#5269]
E:\workspace\terasology1\Terasology>groovyw module init iota
org.codehaus.groovy.control.MultipleCompilationErrorsException: startup failed:
E:\workspace\terasology1\Terasology\config\groovy\common.groovy: 12: unable to resolve class org.eclipse.jgit.errors.RepositoryNotFoundException
 @ line 12, column 1.
   import org.eclipse.jgit.errors.RepositoryNotFoundException
   ^

1 error

How can I solve this problem?

**Comment by BenjaminAmos** [comment]
I do not believe I have seen this particular error before. Have you made sure that you are using Java 17?

It could be that the dependency failed to download. Can you access https://ajoberstar.github.io/bintray-backup/org/ajoberstar/grgit/1.9.3/grgit-1.9.3.pom successfully?

**Comment by zhaohailongzhl** [comment]
> I do not believe I have seen this particular error before. Have you made sure that you are using Java 17?
> 
> It could be that the dependency failed to download. Can you access https://ajoberstar.github.io/bintray-backup/org/ajoberstar/grgit/1.9.3/grgit-1.9.3.pom successfully?

my java version
```
C:\Users\ZHL>java --version
java 17.0.10 2024-01-16 LTS
Java(TM) SE Runtime Environment (build 17.0.10+11-LTS-240)
Java HotSpot(TM) 64-Bit Server VM (build 17.0.10+11-LTS-240, mixed mode, sharing)
```
and I can access the website you said and download it , I also tried java11, but it didn't work

**Comment by soloturn** [comment]
you did something like:

    git clone --filter=blob:none git@github.com:MovingBlocks/Terasology.git 
    cd terasology
    git checkout develop
    groovyw module init iota

is it? or you were using master branch?

**Comment by zhaohailongzhl** [comment]
I clone the develop branch directly. It's default, isn't it? I use develop branch

**Comment by zhaohailongzhl** [comment]
Can I manually init or clone these modules such as https://github.com/Terasology/CoreWorlds ？

**Comment by BenjaminAmos** [comment]
That should be possible What `groovyw` does internally is clone each module internally.

For each module you can do `git clone https://github.com/Terasology/<Module> modules/<Module>`.

The list of `iota` modules can be found at https://github.com/Terasology/Index/blob/master/distros/iota/gradle.properties, which would be:
- BiomesAPI
- CoreAdvancedAssets
- CoreAssets
- CoreRendering
- CoreSampleGameplay
- CoreWorlds
- Drops
- Explosives
- Furnishings
- Health
- Inventory
- ModuleTestingEnvironment

**Comment by zhaohailongzhl** [comment]
I manually clone these modules but can't build them because they don't have the build.gradle.kts file. How do I fix this？

**Comment by BenjaminAmos** [comment]
You can try copying the file [`templates/build.gradle`](https://github.com/MovingBlocks/Terasology/blob/develop/templates/build.gradle) into each cloned module directory. Almost all the modules should use the same `build.gradle` file, as they are typically not allowed to use their own custom one.

**Comment by soloturn** [comment]
clone https://github.com/SiliconSaga/yggdrasil and initialize it with https://github.com/SiliconSaga/realm-siliconsaga/
